src/gs-shell.c

```diff
@@ -599,8 +599,14 @@ on_overview_categories_loaded (GsOverviewPage *page, gpointer user_data)
 					      user_data);
 
 	if (category == NULL) {
-		g_warning ("No %s category after overview categories are refreshed!",
-			   data->category_name);
+		/* this is only an error condition if a non-NULL category name
+		 * is provided; otherwise, selecting the category after creation
+		 * was not intended */
+		if (data->category_name != NULL) {
```

Comment:
This sounds like a different issue that should be addressed in a separate patch. Or is this problem introduced by your patch? The comment also doesn't make it clear for me.

Reply:
I split this out in a separate patch for clarity and cleaned up this comment